src/System.CommandLine/ArgumentArity.cs

```diff
@@ -21,13 +23,18 @@ public ArgumentArity(int minimumNumberOfValues, int maximumNumberOfValues)
                 throw new ArgumentException($"{nameof(maximumNumberOfValues)} must be greater than or equal to {nameof(minimumNumberOfValues)}");
             }
 
+            if (maximumNumberOfValues > MaximumArity)
+            {
+                throw new ArgumentException($"{nameof(maximumNumberOfValues)} must be less than or equal to {nameof(MaximumArity)}");
+            }
+
             MinimumNumberOfValues = minimumNumberOfValues;
             MaximumNumberOfValues = maximumNumberOfValues;
         }
 
-        public int MinimumNumberOfValues { get; set; }
+        public int MinimumNumberOfValues { get; }
 
-        public int MaximumNumberOfValues { get; set; }
+        public int MaximumNumberOfValues { get; }
```

Comment:
I wasn't sure if there was a good reason for making this class mutable. Obviously the setters could be put back and just add checking. I noticed that the other constraints from the constructor were missing on the setters and just decided they were unlikely to be used and removed them

Reply:
Yeah, I'm all for making it immutable.